library/core/src/ptr/mod.rs

```diff
@@ -270,11 +275,11 @@
 //!
 //! But it *is* still sound to:
 //!
-//! * Create an invalid pointer from just an address (see [`ptr::invalid`][]). This can
+//! * Create a provenance-free pointer from just an address (see [`ptr::dangling`][]). This can
```

Comment:
Nit: A "provenance-free pointer" sounds a bit like it's a pointer that isn't bothered by provenance rules. Like a pointer with fewer restrictions, instead of more.

Is there a better, less confusing term maybe?

Reply:
Another reasonable option could be to call it "null provenance" instead (cf. "null object"). With #121242 changing implementation to `null().wrapping_byte_add(addr)` (LLVM `getelementptr` instead of `inttoptr`), it's not even that much of a misnomer to imply it's a pointer with provenance from nullptr.

That also hints that the alternative spelling the method "competes" with is `null().with_addr(addr)`. I still like `dangling_at` but don't hold a strong opinion. Although `null().with_addr` *is* actually fairly clear about semantics (given ZST read is no-op).

Reply:
> LLVM getelementptr instead of inttoptr

It's not currently inttoptr though, it's currently a transmute?

Reply:
> I still like dangling_at

Same. But in the end I don't care what it's called, I basically only care about the semantics.^^ I mostly avoid even getting into discussion that revolve around syntax, and hope someone else picks up those topics, but in this case that strategy doesn't seem to work.